corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -252,20 +263,107 @@ public int printAllProtoDescriptors() {
+     *  record was edited, either due to an error or key not found.
+     */
+    public CorfuDynamicRecord editRecord(String namespace, String tableName,
+        String keyToEdit, String newRecord) {
+        verifyNamespaceAndTableName(namespace, tableName, true);
+
+        if (keyToEdit == null) {
+            throw new IllegalArgumentException("Please specify the key to " +
+                "edit --keyToEdit");
+        }
+
+        if (newRecord == null) {
+            throw new IllegalArgumentException("Please specify the newRecord " +
+                "--newRecord");
+        }
+
+        log.info("\n======================\n");
+        String fullName = TableRegistry.getFullyQualifiedTableName(namespace,
+            tableName);
+        UUID streamUUID = CorfuRuntime.getStreamID(fullName);
+
+        TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
+            .setNamespace(namespace).build();
+
+        CorfuStoreMetadata.KeyValueDefaultRecord keyValueDefaultRecord =
+            dynamicProtobufSerializer.getCachedKeyValueTypesTable().get(tableNameProto);
+
+        Any defaultKeyAny = keyValueDefaultRecord.getKey();
+        Any defaultValueAny = keyValueDefaultRecord.getValue();
+        DynamicMessage keyMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultKeyAny,
+                keyToEdit);
+        DynamicMessage newValueMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultValueAny,
+            newRecord);
+
+        if (keyMsg == null || newValueMsg == null) {
+            return null;
+        }
+
+        runtime.getObjectsView().TXBegin();
+        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table =
+            getTable(namespace, tableName);
+
+        CorfuDynamicKey dynamicKey =
+            new CorfuDynamicKey(defaultKeyAny.getTypeUrl(), keyMsg);
+
+        CorfuDynamicRecord editedRecord = null;
+        if (table.containsKey(dynamicKey)) {
+            CorfuDynamicRecord oldRecord = table.get(dynamicKey);
+
+            if (oldRecord == null) {
+                log.warn("Unexpedted Null Value found for key {} in table " +
+                    "{} and namespace {}.  Stream Id {}", keyToEdit, tableName,
+                    namespace, streamUUID);
+            } else {
+                log.info("Editing record with Key {} in table {} and " +
+                        "namespace {} with new record {}.  Stream Id {}.",
+                    keyToEdit, tableName, namespace, newRecord, streamUUID);
+                String payloadTypeUrl = oldRecord.getPayloadTypeUrl();
+                String metadataTypeUrl = oldRecord.getMetadataTypeUrl();
+                DynamicMessage metadata = oldRecord.getMetadata();
+                editedRecord = new CorfuDynamicRecord(payloadTypeUrl, newValueMsg,
+                    metadataTypeUrl, metadata);
+                table.put(dynamicKey, editedRecord);
```

Comment:
Looks like the new record will have the same metadata as the old one, is it correct?
For example if the record has a revision metadata

Reply:
In all cases I have come across, we want to modify the value as it was incorrectly written.  But you do raise a valid point if we should support editing the metadata.  

Maybe we can address it in a follow-up PR?

Reply:
I just created a GitHub issue to track it